2017-10-27  Kaela Nicole Polnitz  <classkae@lnxsrv09.seas.ucla.edu>
       * othertimespec.h (timespec_cmp): Replaces flawed subtraction operations for tracking
       * time with if-else statements that adequate acquire and list time.


--- timespec.h	2016-10-27 19:23:53.970901000 -0700
+++ othertimespec.h	2017-10-27 19:26:20.816404000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+    if (a.tv_sec > b.tv_sec)
+    {
+        return 1;
+    }
+    else if (a.tv_sec < b.tv_sec)
+    {
+        return -1;
+    }
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
